substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2631,16 +2638,24 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 
 			ensure!(PoolMembers::<T>::contains_key(&who), Error::<T>::PoolMemberNotFound);
-			ClaimPermissions::<T>::mutate(who, |source| {
-				*source = permission;
-			});
+
+			// If the permission is `None`, then remove the claim permission.
+			match permission.default_as_none() {
```

Comment:
So if someone sets `ClaimWithdraw` we will never store it in the state?

Reply:
I see the optimisation and don't see anything wrong here but I feel still better to explicitly store this in the state. Never know some migration bug might make us regret this.

Reply:
Sure. This was a super lean approach of minimising storage but also understand the risk of future migrations or changes to default.